docs/root/configuration/http_filters/lua_filter.rst

```diff
@@ -284,6 +284,16 @@ add()
 Adds a header. *key* is a string that supplies the header key. *value* is a string that supplies
 the header value.
 
+.. attention::
+
+  Envoy treats certain headers specially. These are known as the O(1) or *inline* headers. The
+  list of inline headers can be found `here <https://github.com/envoyproxy/envoy/blob/6b6ce85a24094146c5c225f19b6ecc47b2ca84bf/include/envoy/http/header_map.h#L228>`_.
+  If an inline header is already present in the header map, *add()* will have no effect. If
+  attempting to *add()* a non-inline header, the additional header will be added so that the
+  resultant headers contains multiple header entries with the same name. Consider using the
+  *replace* function if want to replace a header with another value. Note also that we
+  understand this behavior is confusing and we may change it in a future release.
```

Comment:
optional: Might be worth linking to the bug as we expect the behavior to change.
I've commented in the bug to reduce the odds we forget, which is enough for me :-)